packages/smb-client/src/client/broker-client.class.test.ts

```diff
@@ -0,0 +1,166 @@
+  const setWelcomeEvent = context.spies.eventsOn.args.filter((array) => array[0] === 'welcome');
+  assert.is(setWelcomeEvent[0][0], 'welcome');
+
+  const removeWelcomeEvent = context.spies.eventsRemove.args.filter((array) => array[0] === 'welcome');
+  assert.is(removeWelcomeEvent[0][0], 'welcome');
+
+  assert.is(context.spies.socketsSet.calledOnce, true);
+  assert.instance(context.spies.socketsSet.args[0][0], BrokerSocket);
+  assert.is(context.spies.socketsSet.args[0][0].id, 'id_1');
+
+  assert.is(context.spies.socketOn.called, true);
+  assert.is(context.spies.socketOn.args.filter((args) => args[0] === 'data').length, 1);
+  assert.is(clientId, 'id_1');
+});
+
+test('should timeout a connection', async (context) => {
+  const brokerClient = context.clientFactory.createClient();
+
+  try {
+    await brokerClient.connect({ connectionTimeout: 500 });
+  } catch (error) {
+    assert.instance(error, BrokerConnectionTimeoutException);
+    assert.is(context.spies.socketRemoveAllListeners.calledOnce, true);
+    assert.ok(context.spies.socketRemoveAllListeners.calledWith('data'));
+  }
+});
+
+test('should send subscribe command', async (context) => {
+  const subscription: ISubscribeCommand = {
+    command: 'subscribe',
+    action: { topic: 'topic' },
+    socketId: 'id_1',
+    issueTimestamp: Date.now(),
+  };
+
+  const brokerClient = context.clientFactory.createClient();
+  context.welcomeMessage();
+
+  await brokerClient.connect();
+
+  brokerClient.subscribe('topic', context.sandbox.fake());
+  assert.is(context.spies.subsServiceSet.calledOnce, true);
+  assert.is(context.spies.subsServiceSet.lastCall.firstArg, 'topic');
+
+  // Getting the timestamps, which are the only not predictable values.
+  const subscriptionTimestamp = context.spies.eventsNext.args.filter((array) => array[0] === 'subscribe');
+  subscription.issueTimestamp = subscriptionTimestamp[0][1].issueTimestamp;
+  assert.ok(context.spies.eventsNext.calledWith('subscribe', subscription));
+});
+
+test('should send unsubscribe command', async (context) => {
+  const unsubscription: IUnsubscribeCommand = {
+    command: 'unsubscribe',
+    action: { topic: 'topic' },
+    socketId: 'id_1',
+    issueTimestamp: Date.now(),
+  };
+
+  const brokerClient = context.clientFactory.createClient();
+  context.welcomeMessage();
+
+  await brokerClient.connect();
+
+  brokerClient.unsubscribe('topic');
+  assert.is(context.spies.subsServiceUnset.calledOnce, true);
+  assert.is(context.spies.subsServiceUnset.lastCall.firstArg, 'topic');
+
+  // Getting the timestamps, which are the only not predictable values.
+  const unsubscriptionTimestamp = context.spies.eventsNext.args.filter((array) => array[0] === 'unsubscribe');
+  unsubscription.issueTimestamp = unsubscriptionTimestamp[0][1].issueTimestamp;
+  assert.ok(context.spies.eventsNext.calledWith('unsubscribe', unsubscription));
+});
+
+test('should send message command', async (context) => {
+  const message: IMessageCommand = {
```

Comment:
Let's rename to `expectedMessage` so that it's clear that it is something we expect to match rather than a payload we use in the test.